app/src/main/java/org/mozilla/fenix/home/collections/CollectionViewHolder.kt

```diff
@@ -0,0 +1,194 @@
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.compose.list.ExpandableListHeader
+import org.mozilla.fenix.ext.getIconColor
+import org.mozilla.fenix.home.sessioncontrol.CollectionInteractor
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * [RecyclerView.ViewHolder] for displaying an individual [TabCollection].
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param interactor [CollectionInteractor] callback for user interaction.
+ */
+class CollectionViewHolder(
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: CollectionInteractor
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+    private var collection: TabCollection? = null
+    private var isExpanded = false
+
+    @Composable
+    override fun Content() {
+        collection?.let { validCollection ->
+            val horizontalPadding = composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+            composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+
+            val menuItems = getMenuItems(
+                collection = validCollection,
+                interactor = interactor
+            )
+
+            Column {
+                Spacer(Modifier.height(12.dp))
+
+                Collection(
+                    collection = validCollection,
+                    expanded = isExpanded,
+                    menuItems = menuItems,
+                    interactor = interactor
+                )
+            }
+        }
+    }
+
+    /**
+     * Dynamically replace the current [TabCollection] shown in this `ViewHolder`.
+     *
+     * @param collection [TabCollection] to be shown
+     * @param expanded Whether to show the collection as expanded or collapsed.
+     */
+    fun bindSession(collection: TabCollection, expanded: Boolean) {
+        this.collection = collection
+        isExpanded = expanded
+    }
+
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+        const val maxTitleLength = 20
+    }
+}
+
+/**
+ * Rectangular shape with all corners rounded used to display a collapsed collection.
+ */
+private val collapsedCollectionShape = RoundedCornerShape(8.dp)
+
+/**
+ * Rectangular shape with only the top corners rounded used to display an expanded collection with other views
+ * placed immediately below this which can be shown immediately next to it, with no visible separation.
+ */
+private val expandedCollectionShape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp)
+
+@Composable
+private fun Collection(
+    collection: TabCollection,
+    expanded: Boolean,
+    menuItems: List<CollectionMenuItemData>,
+    interactor: CollectionInteractor,
```

Comment:
We should avoid passing the `interactor` into any @Composable functions. Can you add their equivalent `onXXXX` callback function?